src/lib/sql/sql_pipeline_statement.hpp

```diff
@@ -21,6 +21,13 @@ struct SQLPipelineStatementMetrics {
   bool query_plan_cache_hit = false;
 };
 
+enum class SQLPipelineStatus {
+  NotExecuted,  // The pipeline or the pipeline statement has been not been executed yet.
+  Success,      // The pipeline or the pipeline statement has been executed successfully. It is not necessarily
```

Comment:
without info as to why it is "not necessarily committed", this comment will probably make people nervous. (Something called "success" NOT indicating "real" success, but what..?)